### ISSUE TITLE
[BUG] macOS : Finder crashes when you click to open the file at the end of the video downloading

### ISSUE THREAD

**Issue by joyel24** [aandrew-me/ytDownloader#274]
**Info**
 - Operating system: macOS 12.7.6
 - App version: 3.18.5
 - Package version (flatpak/snap/other): No idea, from here https://github.com/aandrew-me/ytDownloader/releases/download/v3.18.5/YTDownloader_Mac_x64.dmg

**Comment by aandrew-me** [comment]
Can you provide the video link

**Comment by joyel24** [comment]
> Can you provide the video link

Yes there it is : https://youtu.be/BwNgWKnhmkg

**Comment by aandrew-me** [comment]
Does your mac have an intel or an arm chip?

**Comment by aandrew-me** [comment]
> > Can you provide the video link
> 
> Yes there it is : https://youtu.be/BwNgWKnhmkg

I see, thanks for such a detailed video

**Comment by aandrew-me** [comment]
Also, can you open the developer tools (Cmd + Shift + I) and then try to do it, see if you see any error messages there

**Comment by joyel24** [comment]
> Does your mac have an intel or an arm chip?

its x64 so intel

**Comment by joyel24** [comment]
> Also, can you open the developer tools (Cmd + Shift + I) and then try to do it, see if you see any error messages there

The shortcut is (alt + Cmd + i) but there is no message in the console when open in finder button is clicked.

**Comment by joyel24** [comment]
I tried to use the console to open the path by looking to your code :
Maybe the location is not mac/linux compatible as it works when I try like in this capture :

<img width="237" alt="Capture d’écran 2024-12-30 à 09 12 07" src="https://github.com/user-attachments/assets/e7477e0b-2f36-4976-a142-5a4e43809680" />

**Comment by aandrew-me** [comment]
> I tried to use the console to open the path by looking to your code : Maybe the location is not mac/linux compatible as it works when I try like in this capture :
> 
> <img alt="Capture d’écran 2024-12-30 à 09 12 07" width="237" src="https://private-user-images.githubusercontent.com/21010316/399248063-e7477e0b-2f36-4976-a142-5a4e43809680.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzU2NDMwODYsIm5iZiI6MTczNTY0Mjc4NiwicGF0aCI6Ii8yMTAxMDMxNi8zOTkyNDgwNjMtZTc0NzdlMGItMmYzNi00OTc2LWExNDItNWE0ZTQzODA5NjgwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMzElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjMxVDEwNTk0NlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTg4MGYyMjVlMjVmZTQ5OTE2YWJmYzAyN2RmNWNiODkyMzQyMjA1ZDRjM2FmNDk3ODcyMDE1MmYxZmU2ODUyZDQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.T-BEP_PLgurGhuCObx1R0-SLCXy6asGG85PFS2DWRvA">

Its just one line of code for all operating systems, and I had tested on Linux and it works fine

**Comment by aandrew-me** [comment]
The only possibility I see is the file path not being correct when its trying to open

**Comment by joyel24** [comment]
> The only possibility I see is the file path not being correct when its trying to open

Yes this was what I expected when I said not mac/linux compatible about the path...

I'll try to change the download directory but it's the default settings actually.

**Comment by joyel24** [comment]
I don't really want to download all source code and dependancies to fix a little bug like this...
Can you just compile for me a version with this little modification and I'll tell you where is the error.

https://github.com/aandrew-me/ytDownloader/blob/02ded1df99598472a3ccc1559185c4afdadfbc48/src/playlist.js#L628


Change this function :
```
function openFolder(location) {
	shell.openPath(location);
}
```
To This:
```
function openFolder(location) {
        console.log(location)
	shell.openPath(location);
}
```

**Comment by aandrew-me** [comment]
The function which you pointed is used in playlists, for opening folders, so you're looking for a different function called `showItem`

 You don't have to download the source code. You can open the console, select render.js, search for the text: `showItem`, and then just add the console log there.
`console.log(path.join(downloadDir, filename + `.${ext}`));`

![image](https://github.com/user-attachments/assets/7b72df40-34aa-40a3-8245-e757aa68b112)

**Comment by aandrew-me** [comment]
But there is already a console.log in code which prints the filepath after the file is saved, so you can check the console and see it

**Comment by aandrew-me** [comment]
If you have the whole filepath, you can try executing
`shell.showItemInFolder("filepath")`

**Comment by aandrew-me** [comment]
If you want, you can message me on telegram - t.me/aandrew_me

**Comment by aandrew-me** [comment]
Were you able to figure anything out?

**Comment by aandrew-me** [comment]
I am making some changes. Check if the issue persists in next update

**Comment by joyel24** [comment]
Okey, yes I'll check that !
I forgot this bug, maybe it's also due to the version of macOS, it's not the latest.

**Comment by aandrew-me** [comment]
Also, after the file is saved, there is a notification and If you clock on it, it should open the file. Does that work for you?

**Comment by joyel24** [comment]
I never tried this way, I often disable desktop notification from apps to prevent spam.
